analyses/independent-samples/01-generate-independent-specimens.R

```diff
@@ -122,6 +138,32 @@ wxs_panel_primary_all <- independent_samples(wxs_panel_samples, tumor_types = "p
 wxs_panel_relapse_all <- independent_samples(wxs_panel_samples, tumor_types = "relapse", independent_level = "all-cohorts", seed = 2020)
 wxs_panel_primary_plus_all <- independent_samples(wxs_panel_samples, tumor_types = "prefer_primary", independent_level = "all-cohorts", seed = 2020)
 
+# Generate WXS independent samples
+wxs_samples <- tumor_samples %>%
+  dplyr::filter(experimental_strategy == "WXS")
+
+wxs_primary_each <- independent_samples(wxs_samples, tumor_types = "primary", independent_level = "each-cohort", seed = 2020)
+wxs_relapse_each <- independent_samples(wxs_samples, tumor_types = "relapse", independent_level = "each-cohort", seed = 2020)
+wxs_primary_plus_each <- independent_samples(wxs_samples, tumor_types = "prefer_primary", independent_level = "each-cohort", seed = 2020)
+
+wxs_primary_all <- independent_samples(wxs_samples, tumor_types = "primary", independent_level = "all-cohorts", seed = 2020)
+wxs_relapse_all <- independent_samples(wxs_samples, tumor_types = "relapse", independent_level = "all-cohorts", seed = 2020)
+wxs_primary_plus_all <- independent_samples(wxs_samples, tumor_types = "prefer_primary", independent_level = "all-cohorts", seed = 2020)
+
+# Generate lists for WGS and Panel samples 
+# We need to generate a separate independent sample list where WXS is preferred - for that list, we need to use WGS+Panel when WXS is not available
+wgs_panel_samples <-  tumor_samples %>% 
+  dplyr::filter(!(Kids_First_Participant_ID %in% 
+                    wxs_samples$Kids_First_Participant_ID))
```

Comment:
```suggestion
wgs_panel_samples <-  tumor_samples %>% 
  dplyr::filter(!(Kids_First_Biospecimen_ID %in% 
                    wxs_samples$Kids_First_Biospecimen_ID))
```

Reply:
And similarly we can make that change for `wxs_panel_samples`

Reply:
This is just so we have more info to discuss in our call: I just added the two suggestions locally i.e. 1. changing `Kids_First_Participant_ID` to `Kids_First_Biospecimen_ID` and 2. adding additional columns to output tsv (for quick debugging purpose) and here is the output with the two participants mentioned in `participants_diff_in_relapse.txt`:

```
$ grep PT_KZ56XHJT *prefer.wxs.tsv

independent-specimens.wgswxspanel.primary-plus.eachcohort.prefer.wxs.tsv:PT_KZ56XHJT	BS_FWP8ZA4K	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary-plus.eachcohort.prefer.wxs.tsv:PT_KZ56XHJT	BS_H8NWA41N	PBTA	WGS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary-plus.prefer.wxs.tsv:PT_KZ56XHJT	BS_FWP8ZA4K	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary-plus.prefer.wxs.tsv:PT_KZ56XHJT	BS_H8NWA41N	PBTA	WGS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.eachcohort.prefer.wxs.tsv:PT_KZ56XHJT	BS_FWP8ZA4K	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.eachcohort.prefer.wxs.tsv:PT_KZ56XHJT	BS_H8NWA41N	PBTA	WGS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.prefer.wxs.tsv:PT_KZ56XHJT	BS_FWP8ZA4K	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.prefer.wxs.tsv:PT_KZ56XHJT	BS_H8NWA41N	PBTA	WGS	Initial CNS Tumor
independent-specimens.wgswxspanel.relapse.eachcohort.prefer.wxs.tsv:PT_KZ56XHJT	BS_9DN4QR6E	PBTA	WXS	Progressive
independent-specimens.wgswxspanel.relapse.eachcohort.prefer.wxs.tsv:PT_KZ56XHJT	BS_D6STCMQS	PBTA	WGS	Progressive Disease Post Mortem
independent-specimens.wgswxspanel.relapse.prefer.wxs.tsv:PT_KZ56XHJT	BS_9DN4QR6E	PBTA	WXS	Progressive
independent-specimens.wgswxspanel.relapse.prefer.wxs.tsv:PT_KZ56XHJT	BS_D6STCMQS	PBTA	WGS	Progressive Disease Post Mortem

$ grep PT_NK8A49X5 *prefer.wxs.tsv

independent-specimens.wgswxspanel.primary-plus.eachcohort.prefer.wxs.tsv:PT_NK8A49X5	BS_1MME7FBS	PBTA	WGS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary-plus.eachcohort.prefer.wxs.tsv:PT_NK8A49X5	BS_6DT506HY	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary-plus.prefer.wxs.tsv:PT_NK8A49X5	BS_1MME7FBS	PBTA	WGS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary-plus.prefer.wxs.tsv:PT_NK8A49X5	BS_6DT506HY	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.eachcohort.prefer.wxs.tsv:PT_NK8A49X5	BS_1MME7FBS	PBTA	WGS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.eachcohort.prefer.wxs.tsv:PT_NK8A49X5	BS_6DT506HY	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.prefer.wxs.tsv:PT_NK8A49X5	BS_1MME7FBS	PBTA	WGS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.prefer.wxs.tsv:PT_NK8A49X5	BS_6DT506HY	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.relapse.eachcohort.prefer.wxs.tsv:PT_NK8A49X5	BS_GKXD33YH	PBTA	WXS	Progressive
independent-specimens.wgswxspanel.relapse.eachcohort.prefer.wxs.tsv:PT_NK8A49X5	BS_HEJ72V3F	PBTA	WGS	Progressive
independent-specimens.wgswxspanel.relapse.prefer.wxs.tsv:PT_NK8A49X5	BS_GKXD33YH	PBTA	WXS	Progressive
independent-specimens.wgswxspanel.relapse.prefer.wxs.tsv:PT_NK8A49X5	BS_HEJ72V3F	PBTA	WGS	Progressive
```

vs before making any changes:

```
$ grep PT_NK8A49X5 *prefer.wxs.tsv

independent-specimens.wgswxspanel.primary-plus.eachcohort.prefer.wxs.tsv:PT_NK8A49X5	BS_6DT506HY	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary-plus.prefer.wxs.tsv:PT_NK8A49X5	BS_6DT506HY	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.eachcohort.prefer.wxs.tsv:PT_NK8A49X5	BS_6DT506HY	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.prefer.wxs.tsv:PT_NK8A49X5	BS_6DT506HY	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.relapse.eachcohort.prefer.wxs.tsv:PT_NK8A49X5	BS_GKXD33YH	PBTA	WXS	Progressive
independent-specimens.wgswxspanel.relapse.prefer.wxs.tsv:PT_NK8A49X5	BS_GKXD33YH	PBTA	WXS	Progressive

$ grep PT_KZ56XHJT *prefer.wxs.tsv

independent-specimens.wgswxspanel.primary-plus.eachcohort.prefer.wxs.tsv:PT_KZ56XHJT	BS_FWP8ZA4K	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary-plus.prefer.wxs.tsv:PT_KZ56XHJT	BS_FWP8ZA4K	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.eachcohort.prefer.wxs.tsv:PT_KZ56XHJT	BS_FWP8ZA4K	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.primary.prefer.wxs.tsv:PT_KZ56XHJT	BS_FWP8ZA4K	PBTA	WXS	Initial CNS Tumor
independent-specimens.wgswxspanel.relapse.eachcohort.prefer.wxs.tsv:PT_KZ56XHJT	BS_9DN4QR6E	PBTA	WXS	Progressive
independent-specimens.wgswxspanel.relapse.prefer.wxs.tsv:PT_KZ56XHJT	BS_9DN4QR6E	PBTA	WXS	Progressive
```

Reply:
```text
wxs_sample_histology_df <- select_wxs(all_samples_histology_df)

wxs_all_cohort_isl <- select_all_cohort_independent_samples(wxs_sample_histology_df)
wxs_each_cohort_isl <- select_each_cohort_independent_samples(wxs_sample_histology_df)

other_all_cohort_isl <- select_all_cohort_independent_samples(
  select(all_samples_histology_df, patient_id not in wxs_sample_histology_df))

other_each_cohort_isl <- select_each_cohort_independent_samples(
  select(all_samples_histology_df,
  specimen_id not in wxs_sample_histology_df,
  sample cohort of patient in wxs_each_cohort_isl is different)
```